docs/specification/new-spec/proposer-selection.md

```diff
@@ -0,0 +1,47 @@
+# Proposer selection procedure in Tendermint
+
+This document specifies the Proposer Selection Procedure that is used in Tendermint to choose a round proposer. 
+As Tendermint is “leader-based protocol”, the proposer selection is critical for its correct functioning.
+Let denote with `proposer_p(h,r)` a process returned by the Proposer Selection Procedure at the process p, at height h
+and round r. Then the Proposer Selection procedure should fulfill the following properties:
+
+`Agreement`: Given a validator set V, and two honest validators,
+p and q, for each height h, and each round r,
+proposer_p(h,r) = proposer_q(h,r)
+
+`Liveness`: In every consecutive sequence of rounds of size K (K is system parameter), at least a 
+single round has an honest proposer. 
+
+`Fairness`: The proposer selection is proportional to the validator voting power, i.e., a validator with more 
+voting power is selected more frequently, proportional to its power. More precisely, given a set of processes 
+with the total voting power N, during a sequence of rounds of size N, every process is proposer in a number of rounds 
+equal to its voting power. 
+
+We now look at a few particular cases to understand better how fairness should be implemented.
+If we have 4 processes with the following voting power distribution (p0,4), (p1, 2), (p2, 3), (p3, 4) at some round r,  
+we have the following sequence of proposer selections in the following rounds:
+
+`p0, p1, p2, p3, p0, p0, p1, p2, p3, p0, p0, p1, p2, p3, p0, p0, p1, p2, p3, p0, etc`
```

Comment:
how does this sequence follow from the given distribution of 4,2,3,4 ?

Reply:
I made typo, it should be (p0,4), (p1, 2), (p2, 2), (p3, 2). I will create a small PR to update this.